.github/workflows/docker.yml

```
@@ -0,0 +1,47 @@
+name: docker
+
+on: [push, pull_request]
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        if: github.event_name == 'push'
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Only build
+        if: github.event_name == 'pull_request'
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          file: ./DockerfileNameRegistry
+          tags: bpfk/agora:name-registry-test
+          build-args: |
+            AGORA_VERSION=HEAD
+
+      -
+        name: Build and push
+        if: github.event_name == 'push'
+        id: docker_build_and_push
+        uses: docker/build-push-action@v2
+        with:
+          push: true
```

You can combine those in one step with:
```suggestion
          push: github.event_name == 'push'
```

---

> You can combine those in one step with:

I wish that worked... and I tried it before... but it seems it always trigger even when I do a pull_request

---

> > You can combine those in one step with:
> 
> I wish that worked... and I tried it before... but it seems it always trigger even when I do a pull_request or they don't trigger even for push (I cannot remember which one)

just confirmed, it doesn't push the image:
https://github.com/bpfkorea/agora/actions/runs/495569212
I have to expicitely set it to 'true'